exercises/bob/bob_test.f90

```diff
@@ -0,0 +1,71 @@
+  call assert_equal("Whoa, chill out!", heyBob("WATCH OUT!"), "shouting")
+  ! Test 3: shouting gibberish
+  call assert_equal("Whoa, chill out!", heyBob("FCECDFCAAB"), "shouting gibberish")
+  ! Test 4: asking a question
+  call assert_equal("Sure.", heyBob("Does this cryogenic chamber make me look fat?"), "asking a question")
+  ! Test 5: asking a numeric question
+  call assert_equal("Sure.", heyBob("You are, what, like 15?"), "asking a numeric question")
+  ! Test 6: asking gibberish
+  call assert_equal("Sure.", heyBob("fffbbcbeab?"), "asking gibberish")
+  ! Test 7: talking forcefully
+  call assert_equal("Whatever.", heyBob("Let's go make out behind the gym!"), "talking forcefully")
+  ! Test 8: using acronyms in regular speech
+  call assert_equal("Whatever.", heyBob("It's OK if you don't want to go to the DMV."), "using acronyms in regular speech")
+  ! Test 9: forceful question
+  call assert_equal("Calm down, I know what I'm doing!", heyBob("WHAT THE HELL WERE YOU THINKING?"), "forceful question")
+  ! Test 10: shouting numbers
+  call assert_equal("Whoa, chill out!", heyBob("1, 2, 3 GO!"), "shouting numbers")
+  ! Test 11: no letters
+  call assert_equal("Whatever.", heyBob("1, 2, 3"), "no letters")
+  ! Test 12: question with no letters
+  call assert_equal("Sure.", heyBob("4?"), "question with no letters")
+  ! Test 13: shouting with special characters
+  call assert_equal("Whoa, chill out!", heyBob("ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!"), "shouting with special characters")
+  ! Test 14: shouting with no exclamation mark
+  call assert_equal("Whoa, chill out!", heyBob("I HATE THE DMV"), "shouting with no exclamation mark")
+  ! Test 15: statement containing question mark
+  call assert_equal("Whatever.", heyBob("Ending with ? means a question."), "statement containing question mark")
+  ! Test 16: non-letters with question
+  call assert_equal("Sure.", heyBob(":) ?"), "non-letters with question")
+  ! Test 17: prattling on
+  call assert_equal("Sure.", heyBob("Wait! Hang on. Are you going to be OK?"), "prattling on")
+  ! Test 18: silence
+  call assert_equal("Fine. Be that way!", heyBob(""), "silence")
+  ! Test 19: prolonged silence
+  call assert_equal("Fine. Be that way!", heyBob("          "), "prolonged silence")
+  ! Test 20: alternate silence
+  !call assert_equal("Fine. Be that way!", heyBob("										"), "alternate silence")
+  ! Test 21: multiple line question
+  call assert_equal("Whatever.", heyBob(""// & 
+    & "Does this cryogenic chamber make me look fat?"// & 
+    & "No."), "multiple line question")
+  ! Test 22: starting with whitespace
+  call assert_equal("Whatever.", heyBob("         hmmmmmmm..."), "starting with whitespace")
+  ! Test 23: ending with whitespace
+  call assert_equal("Sure.", heyBob("Okay if like my  spacebar  quite a bit?   "), "ending with whitespace")
+  ! Test 24: other whitespace
+  !call assert_equal("Fine. Be that way!", heyBob(""// & 
+  !  & "
  !	"), "other whitespace")

```

Comment:
The "enter" key here, should it be (or can it be) `\n` instead, to represent a new line character?  Or is this going to be platform dependent?  Is there a string that we can use to represent "operating system new line", whatever that means for whatever platform?

![newlines_representative_of_OS](https://user-images.githubusercontent.com/16747/63135658-a7a62e00-bf9c-11e9-9104-ee0736f49fd9.png)

Or will <keybd>control-m</keybd> handle this?

Reply:
The control-m is the newline for windows, see https://stackoverflow.com/questions/40082707/what-is-the-difference-between-controlm-character-and-a-r-character-and-how
But getting that to work in this simple example is actually hard in Fortran and I would not let a beginner of fortran go down that rabbit hole. Therefore, I commented out that test manually.